[2024-10-27 16:27:24,507][flwr][WARNING] - Both server and strategy were provided, ignoring strategy
[2024-10-27 16:27:24,507][flwr][INFO] - Starting Flower simulation, config: num_rounds=10, no round_timeout
[2024-10-27 16:27:33,945][flwr][INFO] - Flower VCE: Ray initialized with resources: {'CPU': 8.0, 'memory': 966937806.0, 'node:127.0.0.1': 1.0, 'node:__internal_head__': 1.0, 'object_store_memory': 483468902.0}
[2024-10-27 16:27:33,945][flwr][INFO] - Optimize your simulation with Flower VCE: https://flower.ai/docs/framework/how-to-run-simulations.html
[2024-10-27 16:27:33,945][flwr][INFO] - Flower VCE: Resources for each Virtual Client: {'num_cpus': 4, 'num_gpus': 0.0}
[2024-10-27 16:27:33,979][flwr][INFO] - Flower VCE: Creating VirtualClientEngineActorPool with 2 actors
[2024-10-27 16:27:33,981][flwr][INFO] - [INIT]
[2024-10-27 16:27:33,981][flwr][INFO] - Requesting initial parameters from one random client
[2024-10-27 16:27:39,055][flwr][INFO] - Received initial parameters from one random client
[2024-10-27 16:27:39,055][flwr][INFO] - Starting evaluation of initial global parameters
[2024-10-27 16:27:39,070][flwr][ERROR] - shape '[0, 1]' is invalid for input of size 64
[2024-10-27 16:27:39,081][flwr][ERROR] - Traceback (most recent call last):
  File "C:\Users\Riccardo\miniconda3\envs\flower\Lib\site-packages\flwr\simulation\app.py", line 339, in start_simulation
    hist = run_fl(
           ^^^^^^^
  File "C:\Users\Riccardo\miniconda3\envs\flower\Lib\site-packages\flwr\server\server.py", line 492, in run_fl
    hist, elapsed_time = server.fit(
                         ^^^^^^^^^^^
  File "C:\Users\Riccardo\miniconda3\envs\flower\Lib\site-packages\flwr\server\server.py", line 95, in fit
    res = self.strategy.evaluate(0, parameters=self.parameters)
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Riccardo\miniconda3\envs\flower\Lib\site-packages\flwr\server\strategy\fedavg.py", line 167, in evaluate
    eval_res = self.evaluate_fn(server_round, parameters_ndarrays, {})
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "c:\Users\Riccardo\Desktop\PoliMi\Tesi-Francesco\servers\server_scaffold.py", line 257, in evaluate
    loss, accuracy, f1_score = test_multi(net, testloader, device=device)
                               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "c:\Users\Riccardo\Desktop\PoliMi\Tesi-Francesco\model\multiclassnet.py", line 156, in test_multi
    loss = criterion(output, target.view(0, 1))
                             ^^^^^^^^^^^^^^^^^
RuntimeError: shape '[0, 1]' is invalid for input of size 64

[2024-10-27 16:27:39,089][flwr][ERROR] - Your simulation crashed :(. This could be because of several reasons. The most common are: 
	 > Sometimes, issues in the simulation code itself can cause crashes. It's always a good idea to double-check your code for any potential bugs or inconsistencies that might be contributing to the problem. For example: 
		 - You might be using a class attribute in your clients that hasn't been defined.
		 - There could be an incorrect method call to a 3rd party library (e.g., PyTorch).
		 - The return types of methods in your clients/strategies might be incorrect.
	 > Your system couldn't fit a single VirtualClient: try lowering `client_resources`.
	 > All the actors in your pool crashed. This could be because: 
		 - You clients hit an out-of-memory (OOM) error and actors couldn't recover from it. Try launching your simulation with more generous `client_resources` setting (i.e. it seems {'num_cpus': 4, 'num_gpus': 0.0} is not enough for your run). Use fewer concurrent actors. 
		 - You were running a multi-node simulation and all worker nodes disconnected. The head node might still be alive but cannot accommodate any actor with resources: {'num_cpus': 4, 'num_gpus': 0.0}.
Take a look at the Flower simulation examples for guidance <https://flower.ai/docs/framework/how-to-run-simulations.html>.
